### DISCUSSION TITLE
esbmc++ as it is now (limited but working)

### DISCUSSION THREAD

**Discussion by monicamarcus** [esbmc/esbmc#684] [Q&A]
Hi @kunjsong01 

I understand https://github.com/esbmc/esbmc/tree/master/src/clang-cpp-frontend is not complete, but you must have some limited working esbmc++ model checker. I would still like to try it on some simple C++ examples. Is it possible to use the current version of this esbmc++ ?

Thanks and best regards,
Monica

**Comment by lucasccordeiro** [comment]
@kunjsong01: As we discussed today, can I ask you to share a working version of the new front-end with Monica? I'm sure that @monicamarcus can give you good feedback during the development process of this new C++ front-end.

**Reply by monicamarcus** [reply]
Dear @lucasccordeiro @kunjsong01 

I would certainly be glad to share with you my experience with esbmc++ and esbmc, as well as provide whatever feedback I might have. 

Thanks and best regards,
Monica

**Reply by kunjsong01** [reply]
Hi @monicamarcus, 

Unfortunately, we don't have a working example for clang-cpp-frontend to verify C++ programs involving OOP. Currently I'm trying to add support for verifying class member function calls under PR https://github.com/esbmc/esbmc/pull/686. 

Cheers,
Kunjian

**Reply by monicamarcus** [reply]
@kunjsong01 
Hi Kunjian,

Indeed, I would have liked to try a C++ program involving OOP. But I see you are making nice progress. I also looked at the PR you mentioned and I see all checks were successful for the MacOS version. In fact I am interested in the MacOS version because I use a Mac computer. But I certainly understand that it has to work for both Linux and Windows.

Are you using a certain testing framework to write the tests for your code, or are you just writing tests in C++ ? I also left a comment in the conversation section of the PR.

By the way, I am currently working as an Android developer and using JUnit and Espresso to write my tests.

Cheers,
Monica

**Reply by kunjsong01** [reply]
Hi @monicamarcus,

> Are you using a certain testing framework to write the tests for your code, or are you just writing tests in C++ ? 

We are not using a certain testing framework. We wrote a [Python script](https://github.com/esbmc/esbmc/blob/master/regression/testing_tool.py) to automate regressions in the CI. To add a new regression test, the user just need to add the test program as `main.c` and the corresponding test description files to specify the command line arguments and expected result, e.g. https://github.com/esbmc/esbmc/tree/master/regression/esbmc/github_636_0_false. 

Cheers,
Kunjian
